src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskIntegrationJunkTest.java

```diff
@@ -64,8 +63,15 @@
 import com.mongodb.event.CommandFailedEvent;
 import com.mongodb.event.CommandSucceededEvent;
 
+import com.mongodb.kafka.connect.util.jmx.SourceTaskStatistics;
+
+/**
+ * This class contains tests that are supposed to be unit tests, but because of how these tests were
+ * written originally, they became integration tests as a result of refactoring {@link
+ * MongoSourceConfig}, and need to be completely rewritten to remain unit tests.
+ */
 @ExtendWith(MockitoExtension.class)
-class MongoSourceTaskTest {
+class MongoSourceTaskIntegrationJunkTest {
```

Comment:
On one hand we already have the `MongoSourceTaskIntegrationTest` class, and I did not want to move these tests as they don't belong there. On the other hand I wanted to give the class a name that suggests we should refactor the tests (btw, I have just mentioned this in KAFKA-234). I am open to use a different name, if you have one in mind.

The problems with these tests (even as they were before this PR) are:

- They appear to test multiple things within a single test, which makes some of them multi-page tests (see `testCreatesExpectedCollectionCursor`).
- Probably as a consequence of the previous problem, the tests have to call a magic `resetMocks` method instead of re-creating mocks in `setUp`/`tearDown` or within each test.
- The tests heavily rely on internal methods of the object under test. While it may be because the object under test has too many unrelated behaviors and needs to be divided in multiple objects, I expect that the more likely reason this happened is that `MongoSourceTask` is not unit-testable. While `MongoSourceTask` continues to be not unit-testable, the `StartedMongoSourceTask` that was extracted from it is. These tests need to be refactored to unit-test `StartedMongoSourceTask`.

Reply:
I do not think we should track the status via the name of the test. `MongoSourceTaskIntegrationTestTODONeedsRefactoring` would be better, but more obviously not the right thing. If the problems with the test are not obvious to a future maintainer, we might describe them in a comment, or a Jira ticket if we think we should spend time on this (but, if not now, then when?).

Reply:
> we might describe them in a comment, or a Jira ticket if we think we should spend time on this

KAFKA-234 is such a ticket.

> if not now, then when?

When we decide to do KAFKA-234.

> MongoSourceTaskIntegrationTestTODONeedsRefactoring would be better, but more obviously not the right thing

We have to have a name different from `MongoSourceTaskIntegrationTest` because we already have that class. So despite this being not the right thing, do you want the class to be renamed as `MongoSourceTaskIntegrationTestTODONeedsRefactoring`?

Reply:
Done.